framework/src/systems/NonlinearSystemBase.C

```diff
@@ -2061,24 +2064,25 @@ NonlinearSystemBase::computeNodalBCs(const std::set<TagID> & tags)
       else
         nbc_warehouse = &(_nodal_bcs.getVectorTagsObjectWarehouse(tags, 0));
 
-      for (const auto & bnode : bnd_nodes)
-      {
-        BoundaryID boundary_id = bnode->_bnd_id;
-        Node * node = bnode->_node;
-
-        if (node->processor_id() == processor_id())
+      if (nbc_warehouse->size())
```

Comment:
I think the early check should be enough. What I would do is move the creation of `nbc_warehouse` pointer ahead of your early return check, and so then do 
```c++
if (!nbc_warehouse->size())
  return;
```
And then you don't need the later `if` checks that you've added

Reply:
It would reduce the diff too as a nice byproduct

Reply:
Sounds good.

Reply:
Do you think I can also move `reinitNodeFace` inside the following `if`?